### DISCUSSION TITLE
Use terraform_remote_state in globals

### DISCUSSION THREAD

**Discussion by irenair** [terramate-io/terramate#743] [Q&A] [answered]
How to do this in global?

```
globals {
  obj = {
     value = data.terraform_remote_state.vpc
     number = 10
  }
}
```

I tried using `"data.terraform_remote_state.vpc"` and tm_hcl_expression, but it seems that it can't be used in this situation.

**Comment by katcipis** [comment, marked as answer]
Hi @irenair,

Can't you manipulate "data.terraform_remote_state.vpc" as a string on globals, like this:

```
globals {
  obj = {
     value = "data.terraform_remote_state.vpc"
     number = 10
  }
}
```

And then when you need it to generate code, inside the content block, you use `tm_hcl_expression`, like this:

```
generate_hcl "_test.tf" {
  content {
    state = tm_hcl_expression(global.obj.value)
  }
}
```

This should work and the final generated code will be `data.terraform_remote_state.vpc` (no quotes).

There is some more details about this on #724
